var/spack/repos/builtin/packages/flux-pmix/package.py

```diff
@@ -18,10 +18,11 @@ class FluxPmix(AutotoolsPackage):
     maintainers("grondo")
 
     version("main", branch="main")
+    version("0.4.0", sha256="f7f58891fc9d9a97a0399b3ab186f2cae30a75806ba0b4d4c1307f07b3f6d1bc")
     version("0.3.0", sha256="88edb2afaeb6058b56ff915105a36972acc0d83204cff7f4a4d2f65a5dee9d34")
     version("0.2.0", sha256="d09f1fe6ffe54f83be4677e1e727640521d8110090515d94013eba0f58216934")
 
-    depends_on("flux-core@0.49.0:", when="@0.3.0:")
+    depends_on("flux-core@0.49.0", when="@0.3:0.4")
```

Comment:
okay I tried to combine the advice from @tldahlgren with what you said @grondo let me know if it looks OK.

Reply:
Thanks for figuring this out @vsoch! But, should it be
```suggestion
    depends_on("flux-core@0.49.0", when="@0.3:")
```

To say depends on flux-core 0.49.0 or greater when version is greater then v0.3. Sorry I'm not sure if that's the right syntax.

Reply:
I was trying to say "depends on flux-core 0.49 exactly when in the range of 0.3 to 0.4." and it sounds like we want (for now) exactly 0.49.0 when it's greater than 0.3, and then when that constraint is lifted (e.g., we can use higher versions of flux-core) we can add another rule.

Reply:
Sorry, I should have been more clear. v0.4.0 of flux-pmix doesn't depend on _exactly_ flux-core v0.49.0, it (and future flux-pmix releases) depend on _at least_ flux-core v0.49.0. That is, the latest tag v0.55.0 will also work, but v0.48.0 will not. Hope that makes sense, and apologies if that is what is already checked here!